### PR TITLE
fix: Non standard partition fixes

### LINKED ISSUE #873: assume role wrong
### Describe the bug

The arn of role is like to "arn:\<partition\>:iam::\<account id\>:role/<rolename>".
So in China, it should be "arn:aws-cn:iam::1234567890:role/<rolename>", it is "arn:aws:iam::1234567890:role/<rolename>" now.

### Expected Behavior

assume success

### Steps to Reproduce

use org fetch in AWS China

### Possible Solution

_No response_

### Provider and CloudQuery version

Cloudquery: 0.22.10. AWS provider: 0.11.4

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by disq**
Fixes #873 and potentially other issues

### CODE REVIEW

File: client/organizations.go [hunk 1]
```diff
@@ -58,9 +59,20 @@ func loadAccounts(ctx context.Context, awsConfig *Config, accountsApi Organizati
 		if account.Status != orgTypes.AccountStatusActive {
 			continue
 		}
+		roleArn := arn.ARN{
+			Partition: "aws",
```
Comment (bbernays) [inline client/organizations.go:63]: I think the partition can't be hardcoded

Comment (disq) [reply client/organizations.go:63]: we parse and replace it below.

**Review by bbernays — COMMENTED** [review]
Just one comment about naming

**Review by irmatov — APPROVED** [review]
LGTM. Good suggestion by Ben to change naming of Universal/Global.

**Review by roneli — APPROVED** [review]

**Review by spangenberg — APPROVED** [review]
LGTM

**Review by bbernays — APPROVED** [review]

**Pull request merged**